index.js

```diff
@@ -19,48 +16,92 @@ module.exports = (input, options) => {
   options = Object.assign({}, options)
 
   let table = ''
-  let titles = []
-  let alignments = []
 
   let keys = Object.keys(input[0])
 
-  if (!options.columns || !Array.isArray(options.columns)) {
-    // use the keys from the first object
-    titles = keys
-    alignments = new Array(titles.length).fill('')
-  } else {
-    titles = keys.map((key, i) => {
-      let column = options.columns[i]
-
-      if (isPlainObject(column)) {
-        alignments[i] = column.align
-        return column.name || key
-      } else {
-        alignments[i] = ''
-        return column || key
-      }
-    })
-  }
+  let titles = keys.map((key, i) => {
+    if (Array.isArray(options.columns) && options.columns[i]) {
+        if (typeof options.columns[i] === 'string') {
+          return options.columns[i]
+        } else if (options.columns[i].name) {
+          return options.columns[i].name
+        }
+    }
+
+    return key
+  })
+
+  let widths = input.reduce(
+    (sizes, item) => keys.map(
+      (key, i) => Math.max(
+        columnWidthMin,
+        typeof item[key] === 'undefined' ? 0 : String(item[key]).length,
+        sizes[i]
+      )
+    ),
+    titles.map(t => t.length)
+  )
+
+  let alignments = keys.map((key, i) => {
+    if (Array.isArray(options.columns)
+      && options.columns[i]
+      && options.columns[i].align
+    ) {
+      let align = String(options.columns[i].align).toUpperCase()
+
+      if (!ALIGN.includes(align)) throw new TypeError(
```

Comment:
Let's use curly braces here.

```js
if (!ALIGN.includes(align)) {
  throw new TypeError(`Unknown alignment, got ${options.columns[i].align}`)
}
```

Reply:
This is a difference from my usual style, and though I tried to be consistent with yours, this one slipped.

Reply:
All good, it's a nitpick. I wouldn't mind if it weren't for the `throw` spanning multiple lines.

Reply:
Ah I forgot about `Array#includes` - for node 4 support we'll have to use the ol' `indexOf`. I think that's the last thing here.